akka-http-core/src/main/scala/akka/http/javadsl/ConnectHttp.scala

```diff
@@ -66,13 +67,13 @@ object ConnectHttp {
   @throws(classOf[IllegalArgumentException])
   def toHostHttps(host: String, port: Int): ConnectWithHttps = {
     require(port > 0, "port must be > 0")
-    val start = if (host.startsWith("https://")) host else s"https://$host"
+    val start = if (host.startsWith("http://") || host.startsWith("https://")) host else s"https://$host"
```

Comment:
also, a lot of repeated identical string comparisons here, couldn't those be extracted to one method instead?